app/controllers/api/base_controller.rb

```diff
@@ -20,6 +20,10 @@ def authenticate!
       render(status: :unauthorized, json: { error: "Unauthorized" })
     end
 
+    def provider
```

Comment:
Could this be called `current_provider` so that it makes it a bit more obvious that it's the authenticated provider?